projects/igniteui-angular/src/lib/services/overlay/overlay.ts

```diff
@@ -66,35 +69,45 @@ export class IgxOverlayService {
     /**
      * Shows the provided component.
      */
-    show(component: ElementRef | Type<{}>, settings?: OverlaySettings): string {
-        const id: string = (this._componentId++).toString();
-        settings = Object.assign({}, this._defaultSettings, settings);
+    show(component: ElementRef | Type<{}>, settings?: OverlaySettings, id?: string): string {
```

Comment:
Alright, alternative idea for the ID. Feels way more natural to go `show(id, options)` if you already have the ID, no?
And since you'll be making a check for the extra param anyway, you might as well be checking it for an overload.
Try [this on the Typescript playground](https://www.typescriptlang.org/play/index.html#src=class%20OverlaySettings%20%7B%20%7D%0D%0Aclass%20ElementRef%20%7B%20%7D%0D%0Aclass%20Type<T>%20%7B%7D%0D%0A%0D%0Aclass%20Test%20%7B%0D%0A%20%20%20%20show(id%3A%20string%2C%20settings%3F%3A%20OverlaySettings)%3A%20string%3B%0D%0A%20%20%20%20show(component%3A%20ElementRef%20%7C%20Type<%7B%7D>%2C%20settings%3F%3A%20OverlaySettings)%3A%20string%3B%0D%0A%20%20%20%20show(compOrId%3A%20ElementRef%20%7C%20Type<%7B%7D>%20%7C%20string%2C%20settings%3F%3A%20OverlaySettings)%3A%20string%20%7B%0D%0A%20%20%20%20%20%20%20%20if%20(typeof%20compOrId%20%3D%3D%3D%20"string")%20%7B%0D%0A%20%20%20%20%20%20%20%20%20%20%20%20console.log('string%3A'%2C%20compOrId)%3B%0D%0A%20%20%20%20%20%20%20%20%7D%20else%20%7B%0D%0A%20%20%20%20%20%20%20%20%20%20%20%20console.log('Ref%2FType%3A'%2C%20compOrId)%3B%0D%0A%20%20%20%20%20%20%20%20%7D%0D%0A%20%20%20%20%20%20%20%20%0D%0A%20%20%20%20%20%20%20%20return%20"id"%3B%0D%0A%20%20%20%20%7D%0D%0A%7D%0D%0A%0D%0Alet%20test%20%3D%20new%20Test()%3B%0D%0Atest.show("text"%2C%20%7B%7D)%3B%0D%0Atest.show(new%20ElementRef())%3B) - seems to work super nice and the intellisense finds the overloads.